### PR TITLE
add ped phase mapping: Ref #170

### PR DESCRIPTION
**Pull request by adev4a**
# PR Details
This PR adds functionality to the traffic signal controller service to map pedestrian phases to channels.
## Description
Currently the traffic signal controller service creates a map between channel numbers (referred to as signal group id in J2735) and phase numbers. In order to process all incoming SPaT messages, a map is also needed between pedestrian phases and the associated signal group ID.

#170 
## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: tsc_client_service/include/monitor_tsc_state.h [hunk 1]
```diff
@@ -3,43 +3,136 @@
...
+
+        /* Mapping between signal group ids(key) and their states(value) defined as a signal group state struct*/
+        std::unordered_map<int, signal_group_state> signal_group_state_map_;
+
         /* Map between phase numbers(key) and signal group ids(value) for all active vehicle phases in the Traffic Signal Controller*/
-        std::unordered_map<int,int> phase_num_map_;
+        std::unordered_map<int,int> vehicle_phase_num_map_;
 
-        /* Map between signal group ids(key) and phase numbers(value) for all active vehicle phases in the Traffic Signal Controller*/
-        std::unordered_map<int,int> signal_group_map_;
```
Comment (paulbourelly999) [inline tsc_client_service/include/monitor_tsc_state.h:17]: So we removed the signal group to phase number map?

Comment (adev4a) [reply tsc_client_service/include/monitor_tsc_state.h:17]: No, this is just renamed to signal_group_phase_map_

File: tsc_client_service/include/monitor_tsc_state.h [hunk 2]
```diff
@@ -52,15 +55,18 @@ namespace traffic_signal_controller_service
 
         /** @brief Returns a vector of channels associated with a vehicle phase. Ignores pedestrian phase, overlap, ped Overlap, queueJump and other (types defined in NTCIP1202 v03)
```
Comment (paulbourelly999) [inline tsc_client_service/include/monitor_tsc_state.h:56]: Update this part of the documentation

File: tsc_client_service/include/monitor_tsc_state.h [hunk 3]
```diff
@@ -52,15 +55,18 @@ namespace traffic_signal_controller_service
...
-        ** @param vehicle_phase_channels a vector of channel numbers in the traffic signal controller associated with a vehicle phase
+        ** @param phase_channels a vector of channel numbers in the traffic signal controller associated with a phase
+        ** @param is_source_vehicle_channel a boolean that indicates whether the phase_channels arg is for channels associated with vehicle phases. 
+        ** In case they are not its assumed that they are pedestrian phases
         * According to the NTCIP 1202 v03 documentation signal group ids in the SPAT message are the channel numbers in the TSC
         * **/
-        void map_phase_and_signalgroup(const std::vector<int>& vehicle_phase_channels);
+        void map_phase_and_signalgroup(const std::vector<int>& phase_channels, bool is_source_vehicle_channel);
 
         /**
```
Comment (paulbourelly999) [inline tsc_client_service/include/monitor_tsc_state.h:71]: Please add 2 separate getters for each phase number to signal group map. One for vehicle phases and one for pedestrian phases.

**Review by paulbourelly999 — APPROVED** [review]
Looks good. Merging now.

**Pull request merged**